chord_metadata_service/mohpackets/tests/test_models.py

```diff
@@ -0,0 +1,1483 @@
+        self.assertEqual(self.donor.date_of_birth, "1975-08")
+        self.assertEqual(self.donor.date_of_death, "2009-08")
+        self.assertCountEqual(self.donor.primary_site, [
+            "Adrenal gland",
+            "Other and ill-defined sites in lip, oral cavity and pharynx"
+        ])
+
+    # These tests will be implemented after we decide which fields are required
+    # def test_optional_fields(self):
+    #     optional_fields = ["submitter_donor_id", "is_deceased", "cause_of_death",
+    #     "date_of_birth", "date_of_death", "primary_site"]
+    #     for field in optional_fields:
+    #         setattr(self.donor, field, None)
+    #     self.donor.save()
+
+    def test_unique_id(self):
+        with self.assertRaises(IntegrityError):
+            self.donor = Donor.objects.create(**self.valid_values)
+
+    def test_invalid_id(self):
+        invalid_values = get_invalid_ids()
+        for value in invalid_values:
+            with self.subTest(value=value):
+                self.valid_values["submitter_donor_id"] = value
+                self.serializer = DonorSerializer(instance=self.donor, data=self.valid_values)
+                self.assertFalse(self.serializer.is_valid())
+
+    def test_invalid_is_deceased(self):
+        self.donor.is_deceased = "foo"
+        with self.assertRaises(ValidationError):
+            self.donor.save()
+
+    def test_invalid_cause_of_death(self):
+        invalid_values = get_invalid_choices()
+        for value in invalid_values:
+            with self.subTest(value=value):
+                self.valid_values["cause_of_death"] = value
+                self.serializer = DonorSerializer(instance=self.donor, data=self.valid_values)
+                self.assertFalse(self.serializer.is_valid())
+
+    def test_invalid_date_of_death(self):
+        invalid_values = get_invalid_dates()
+        for value in invalid_values:
+            with self.subTest(value=value):
+                self.valid_values["date_of_death"] = value
+                self.serializer = DonorSerializer(instance=self.donor, data=self.valid_values)
+                self.assertFalse(self.serializer.is_valid())
+
+    def test_invalid_date_of_birth(self):
+        invalid_values = get_invalid_dates()
+        for value in invalid_values:
+            with self.subTest(value=value):
+                self.valid_values["date_of_birth"] = value
+                self.serializer = DonorSerializer(instance=self.donor, data=self.valid_values)
+                self.assertFalse(self.serializer.is_valid())
+
+    def test_invalid_primary_site(self):
+        invalid_values = ["foo", ["foo"]]
+        for value in invalid_values:
+            with self.subTest(value=value):
+                self.valid_values["primary_site"] = value
+                self.serializer = DonorSerializer(instance=self.donor, data=self.valid_values)
+                self.assertFalse(self.serializer.is_valid())
```

Comment:
while the invalid test serves it purpose, I think you should add the test for valid primary_site choices also (and for all the similar tests). Otherwise, the test won't confirm anything. 

You can make another list of valid choices (please don't "link" to the permissble_values since we need to double-check any mistakes) Yes, it would mean that if we add new values, we need to change in the test too, but otherwise we won't know if we add something go wrong

Reply:
The test_model_creation tests include a valid value for every single field, and the data is not linked to permissible values. The object is called valid_values.

Reply:
I'm not sure what you mean, the test_model_creation tests include valid values for every single field and they are not linked to permissible values. The object is called self.valid_values.

Reply:
I mean you can make a list of like `get_valid_primary_choices` and assert them as True